### PR TITLE
Fix ext_grains and ext_modules tests for Windows

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Fixes the `test_grains_overwrite` test in `integration.loader.test_ext_grains` tests
Fixes the `test_overridden_internal` test in `integration.loader.test_ext_modules` tests
Adds the tests to the Windows Whitelist
Modifies the names of all the renamed tests in the Windows whitelist
Fixes unreachable `log.debug in in the `grains` function in `loader.py`

### Tests written?
Yes

### CODE REVIEW

File: tests/integration/loader/test_ext_grains.py [hunk 1]
```diff
@@ -30,6 +31,18 @@ class LoaderGrainsTest(integration.ModuleCase):
     #    self.opts['grains'] = grains(self.opts)
 
     def test_grains_overwrite(self):
+        # To avoid a race condition on Windows, we need to make sure the
```
Comment (cachedout) [inline tests/integration/loader/test_ext_grains.py]: This needs more investigation IMHO. What's supposed to be putting this file in the cache? Why is it taking so long? I very much dislike just putting in timeouts when it's not clear what the problem is.

**Review by cachedout — CHANGES_REQUESTED** [review]

File: tests/integration/loader/test_ext_grains.py (hunk 1 above)
Comment (twangboy) [inline tests/integration/loader/test_ext_grains.py]: I don't know...

Comment (cachedout) [inline tests/integration/loader/test_ext_grains.py]: That's a much larger problem then. Let's try to figure out why that might be happening. Let me know if you would like a second pair of eyes to help diagnose the issue there.

Comment (twangboy) [inline tests/integration/loader/test_ext_grains.py]: I don't even know where to start

Comment (cachedout) [inline tests/integration/loader/test_ext_grains.py]: OK. Grab me tomorrow afternoon and we'll have a look. :]

**Comment by twangboy** [comment]
@cachedout I clarified the comments on this one

**Review by cachedout — APPROVED** [review]

**Pull request merged**